### PR TITLE
Port stateroot changes to master

### LINKED ISSUE #1377: Port stateroot changes to master
3.0 needs this too. At the moment we're porting them as is, but we'll try to improve upon them later.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1377 .

Changes:
1. `stateRootEnabled` is no longer used.
2. Items are stored by contract ID.
3. There are now 2 P2P commands instead of 3.

Testing done: `go test`, `make env_up`.

### CODE REVIEW

**Review by roman-khimov — CHANGES_REQUESTED** [review]
It looks good, but maybe it's better to wait a little for neo-project to decide how it really is gonna look like for Neo 3.

File: pkg/rpc/server/server.go [hunk 1]
```diff
@@ -696,6 +697,28 @@ func (s *Server) contractIDFromParam(param *request.Param) (int32, *response.Err
 	return result, nil
 }
 
+func (s *Server) getStateRoot(ps request.Params) (interface{}, *response.Error) {
+	p := ps.Value(0)
+	if p == nil {
+		return nil, response.NewRPCError("Invalid parameter.", "", nil)
```
Comment (AnnaShaleva) [inline pkg/rpc/server/server.go:777]: `return nil, response.ErrInvalidParams`?

File: pkg/rpc/client/rpc_test.go [hunk 2]
```diff
@@ -508,6 +508,27 @@ var rpcClientTestCases = map[string][]rpcClientTestCase{
 			},
 		},
 	},
+	"getstateroot": {
```
Comment (AnnaShaleva) [inline pkg/rpc/client/rpc_test.go:546]: A test for error case can also be added to `rpcClientErrorCase` in order to cover error after `c.PerformRequest`.

**Review by AnnaShaleva — APPROVED** [review]

**Comment by roman-khimov** [comment]
#1500 was merged making this really obsolete. Any new Neo 3 changes wrt this topic are to be implemented in separate PRs based on current master branch.

**Pull request closed without merging**